openqabot/incrementapprover.py

```diff
@@ -171,16 +171,31 @@ def _request_openqa_job_results(
         params: list[dict[str, str]],
     ) -> list[dict[str, dict[str, dict[str, Any]]]]:
         log.debug("Checking openQA job results for %s", build_info)
-        query_params = (
-            {
-                "distri": p["DISTRI"],
-                "version": p["VERSION"],
-                "flavor": p["FLAVOR"],
-                "arch": p["ARCH"],
-                "build": p["BUILD"],
-            }
-            for p in params
-        )
+        query_params = []
+        for p in params:
+            if p["FLAVOR"]:
+                query_params += [
+                    {
+                        "distri": p["DISTRI"],
+                        "version": p["VERSION"],
+                        "flavor": p["FLAVOR"],
+                        "arch": p["ARCH"],
+                        "build": p["BUILD"],
+                    }
+                ]
+            else:
+                for c in self.config:
+                    flavor = c.settings.get("FLAVOR", None)
+                    if flavor:
+                        query_params += [
+                            {
+                                "distri": p["DISTRI"],
+                                "version": p["VERSION"],
+                                "flavor": flavor,
+                                "arch": p["ARCH"],
+                                "build": p["BUILD"],
+                            }
+                        ]
```

Comment:
How about a base dict reused for the flavor/not-flavor branch

```suggestion
for p in params:
    base = {
        "distri": p["DISTRI"],
        "version": p["VERSION"],
        "arch": p["ARCH"],
        "build": p["BUILD"],
    }
    flavors = [p["FLAVOR"]] if p.get("FLAVOR") else [c.settings.get("FLAVOR") for c in self.config if c.settings.get("FLAVOR")]
    query_params.extend({**base, "flavor": f} for f in flavors)
```

I am not sure about the branch when p["FLAVOR"] isn't set and also c.settings.get would not return a value but just "None" though. But I can re-check my suggestions after you added tests :)

Reply:
Certainly nicer than my approach. I will try to come up with better tests to cover this